src/DataProvider/ChainItemDataProvider.php

```diff
@@ -44,7 +44,17 @@ public function getItem(string $resourceClass, $id, string $operationName = null
                     continue;
                 }
 
-                return $dataProvider->getItem($resourceClass, $id, $operationName, $context);
+                $identifier = $id;
+                if (!$dataProvider instanceof DenormalizedIdentifiersAwareItemDataProviderInterface && $identifier && \is_array($identifier)) {
+                    if (\count($identifier) > 1) {
+                        @trigger_error(sprintf('Receiving "$id" as non-array in an item data provider is deprecated in 2.3 in favor of implementing "%s".', DenormalizedIdentifiersAwareItemDataProviderInterface::class), E_USER_DEPRECATED);
+                        $identifier = http_build_query($identifier, '', ';');
+                    } else {
+                        $identifier = current($identifier);
```

Comment:
Update done @dunglas, do you want me to move this statement up to avoid the `else` ?